### PR TITLE
Fix API field references for autoscaling v1, v2 and v2beta2

### PR DESCRIPTION
**Pull request by rayowang**
What type of PR is this?
/kind documentation

Which issue(s) this PR fixes:
#113438

Does this PR introduce a user-facing change?

The PR Changes API field references on API docs. It uses json field names instead of Golang type names.

### CODE REVIEW

File: pkg/apis/autoscaling/types.go [hunk 1]
```diff
@@ -62,31 +62,36 @@ type ScaleStatus struct {
...
-	// Kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
+	// kind is the kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
 	Kind string
-	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
+
+	// name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
 	Name string
-	// API version of the referent
+
+	// APIVersion is the API version of the referent
```
Comment (cici37) [inline pkg/apis/autoscaling/types.go:71]: `apiVersion`?

File: staging/src/k8s.io/api/autoscaling/v1/types.go [hunk 2]
```diff
@@ -46,32 +48,34 @@ type HorizontalPodAutoscalerSpec struct {
...
 	MinReplicas *int32 `json:"minReplicas,omitempty" protobuf:"varint,2,opt,name=minReplicas"`
-	// upper limit for the number of pods that can be set by the autoscaler; cannot be smaller than MinReplicas.
+
+	// maxReplicas is the upper limit for the number of pods that can be set by the autoscaler; cannot be smaller than MinReplicas.
 	MaxReplicas int32 `json:"maxReplicas" protobuf:"varint,3,opt,name=maxReplicas"`
-	// target average CPU utilization (represented as a percentage of requested CPU) over all the pods;
-	// if not specified the default autoscaling policy will be used.
+
+	// targetCPUUtilizationPercentage is the target average CPU utilization (represented as a percentage of requested CPU) over all the pods;
+	// if not specified the default autoscaling pomlicy will be used.
```
Comment (cici37) [inline staging/src/k8s.io/api/autoscaling/v1/types.go:56]: `pomlicy` -> `policy`?

**Review by cici37 — COMMENTED** [review]
/ok-to-test

**Comment by rayowang** [comment]
I have run the pull-kubernetes-verify job many times but failed, and the error is as follows：
`{Script Error ScriptError go version go1.20.1 linux/amd64
rm: cannot remove '/home/prow/go/src/k8s.io/kubernetes/hack/../api/openapi-spec/v3/*': No such file or directory
rm: refusing to remove '.' or '..' directory: skipping '/home/prow/go/src/k8s.io/kubernetes/hack/../api/openapi-spec/v3/.'
rm: refusing to remove '.' or '..' directory: skipping '/home/prow/go/src/k8s.io/kubernetes/hack/../api/openapi-spec/v3/..'
/home/prow/go/src/k8s.io/kubernetes/api/openapi-spec is out of date. Please run hack/update-openapi-spec.sh}`
{Script Error ScriptError go version go1.20.1 linux/amd64

When I run hack/update-openapi-spec.sh, the following error still occurs:
`    hack/../api/openapi-spec for OpenAPI v2
+++ [0224 15:45:32] Updating 
    hack/../api/openapi-spec/v3 for OpenAPI v3
rm: refusing to remove '.' or '..' directory: skipping 'hack/../api/openapi-spec/v3/.'
rm: refusing to remove '.' or '..' directory: skipping 'hack/../api/openapi-spec/v3/..'
+++ [0224 15:45:32] Updating OpenAPI spec for group .well-known/openid-configuration
+++ [0224 15:45:32] Updating OpenAPI spec for group api
`

I would like to ask what is the reason for this? How should it be fixed? thanks @cici37 @ncdc

**Comment by liggitt** [comment]
the `api/openapi-spec/v3/apis__autoscaling__v2beta2_openapi.json` file should not be added...

make sure you fetch tags before you run `make update` so the API docs for the correct version are generated

**Comment by rayowang** [comment]
> the `api/openapi-spec/v3/apis__autoscaling__v2beta2_openapi.json` file should not be added...
> 
> make sure you fetch tags before you run `make update` so the API docs for the correct version are generated

Thx, with your suggestion, the problem mentioned above is fixed. Please help to review again. @cici37 @liggitt

**Comment by liggitt** [comment]
/lgtm
/approve

thanks for the PR!

**Comment by rayowang** [comment]
@liggitt  Hello, I am doing some work related to kubernetes recently, so I want to learn more about the project by solving some issues. But I found that because I am not a member, I cannot set ok-to-test for CI automated testing after submitting the PR, which often blocks my progress and reduces my efficiency.

I read the requirements for membership application, and I basically meet the requirements.
but according to the organization's application for membership, I need to communicate in advance with a sponsor associated with my community contribution. Thank you for your help in the past, I hope you will agree to sponsor my membership application, can you help me?

**Pull request merged**